### PR TITLE
[incubator-kie-drools-6180] accumulate min doesn't evaluate correctly…

### PR DESCRIPTION
**Pull request by tkobayas**
… with more than 18 digits BigDecimal

**Issue**:

* https://github.com/apache/incubator-kie-drools/issues/6180

At first, I thought I can use `MinAccumulateFunction` for `BigDecimal` and `BigInteger`. However, its `getResultType()` returns `Comparable.class`, so it causes a compile error when using methods like `$min.intValue()`. We may change the strategy to cache accumulate function instances (currently, per function name) to make `MinAccumulateFunction` work for various types, but I followed the current approach which is easier to understand, so I added 4 function classes `BigDecimalMinAccumulateFunction` (minBD), `BigIntegerMinAccumulateFunction` (minBI), `BigDecimalMaxAccumulateFunction` (maxBD), `BigIntegerMaxAccumulateFunction` (maxBI)

### CODE REVIEW

**Review by mariofusco — APPROVED** [review]

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java [hunk 1]
```diff
@@ -3941,4 +3943,156 @@ public void testPeerCollectWithEager(KieBaseTestConfiguration kieBaseTestConfigu
...
+    void minWithBigIntegerHighAccuracy(KieBaseTestConfiguration kieBaseTestConfiguration) {
+        final String drl =
+                "import " + Primitives.class.getCanonicalName() + ";\n" +
+                        "global java.util.List results;\n" +
+                        "rule R1 when\n" +
+                        "    accumulate(Primitives($bi : bigInteger), $min : min($bi))\n" +
+                        "then\n" +
+                        "    results.add($min);\n" +
+                        // to confirm if $min is BigInteger at build time (Not Comparable)
+                        // The return value isn't important to assert
```
Comment (pibizza) [inline drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java]: Are these comments relevant? it seems that we are asserting the result values.

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java [hunk 2]
```diff
@@ -3941,4 +3943,156 @@ public void testPeerCollectWithEager(KieBaseTestConfiguration kieBaseTestConfigu
...
+                        "    System.out.println($min.nextProbablePrime());\n" +
```
Comment (pibizza) [inline drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java]: Do we need to printout this value? if possible a test should not print values, only check them in assertions.

**Review by pibizza — COMMENTED** [review]
Check comments to see if they are relevants for the tests.

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java (hunk 2 above)
Comment (tkobayas) [reply drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java]: `nextProbablePrime()` is specific to `BigInteger` class. So, this line is to check that drl build doesn't raise a compile error, so we can confirm that `$min` is treated as `BigInteger` in the internally generated java code. The return value of  `nextProbablePrime()`  ~~is not guaranteed, so we cannot assert the value~~ is not important, I didn't assert.

Having said that, I can avoid `System.out.println` anyway. I'll fix the PR. Thanks.

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java (hunk 1 above)
Comment (tkobayas) [reply drools-test-coverage/test-compiler-integration/src/test/java/org/drools/compiler/integrationtests/AccumulateTest.java]: The comment is to explain the below code `"    System.out.println($min.nextProbablePrime());\n" `. I'll rewrite it better. Thanks.

**Comment by tkobayas** [comment]
re-running GHA

**Comment by tkobayas** [comment]
Test failures on CallbackStateTimeoutsIT, MultipleTimerInstancesIT etc. are being solved by https://github.com/apache/incubator-kie-kogito-runtimes/pull/3814

**Pull request merged**